### PR TITLE
Bugfix/rdmp 136 chi columns

### PR DESCRIPTION
**Pull request by JFriel**
## Changes
* Updates tests to the latest version of NUnit provided by RDMP
* Adds documentation for the CHI Coulmn finder
* Adds a sample AllowList for the CHI Column Finder
* Hardens the CHI Column Finder to not crash when parsing junk CHIs
*Update FoundChi extraction location to be the root extraction folder, not the specific extraction folder

## Solution
A user was passing UIDs into the CHI column finder, such as "1896911d-7148-4bc2-8dc2-ddb371240550", which contains a 9 character substring that looks like a CHI. This caused us to try and grab the 10 digit CHI from String[string.length-9,String.Length+1] which caused the CHI column finder to crash. Some hardening has been done to prevent attempting to pull substrings that are out of bounds

## How to test
You can set up and extraction with the CHI column finder and attempt to extract quasi-CHIs in an attempt to break it, but it will be quicker and easier to add any values you want to testy into the CHI list in the CHIColumnFinderTests.cs file

### CODE REVIEW

File: HICPlugin/DataFlowComponents/CHIColumnFinder.cs [hunk 1]
```diff
@@ -46,6 +46,7 @@ public sealed partial class CHIColumnFinder : IPluginDataFlowComponent<DataTable
 
     private const string RdmpAll = "RDMP_ALL";
     private readonly string _potentialChiLocationFileDescriptor = "_Potential_CHI_Locations.csv";
+    private string ExtractionIdentifier;
```
Comment (bpeacock001) [inline HICPlugin/DataFlowComponents/CHIColumnFinder.cs:49]: The string case needs put to camel case.

File: HICPlugin/DataFlowComponents/CHIColumnFinder.cs [hunk 2]
```diff
@@ -141,13 +145,14 @@ public DataTable ProcessPipelineData(DataTable toProcess, IDataLoadEventListener
             }
         }
 
-        if (count>0 && OutputFileDirectory?.Exists == true)
+        if (count > 0 && OutputFileDirectory?.Exists == true)
         {
             var countWritten = BailOutAfter > 0 ? Math.Min(count, BailOutAfter) : count;
-            listener.OnNotify(this, new NotifyEventArgs(ProgressEventType.Information, $"{countWritten} CHIs have been found in your extraction. Find them in {OutputFileDirectory.FullName}"));
+            var location = (OutputFileDirectory.FullName );
```
Comment (bpeacock001) [inline HICPlugin/DataFlowComponents/CHIColumnFinder.cs:151]: I don't think you need the brackets.

File: HICPluginTests/Integration/ReflectionToDatabaseTester.cs [hunk 3]
```diff
@@ -58,7 +58,7 @@ public void SendTooLongFieldToDatabase()
         try
         {
             var ex = Assert.Throws<Exception>(() => ReflectionBasedSqlDatabaseInserter.MakeInsertSqlAndExecute<TestObject>(t, con, dbInfo, "TestObject"));
-            Assert.IsTrue(ex?.Message.Contains("Field1 in table TestObject is defined as length  10 in the database but you tried to insert a string value of length 41"));
+            Assert.That(ex?.Message.Contains("Field1 in table TestObject is defined as length  10 in the database but you tried to insert a string value of length 41"),Is.True);
```
Comment (bpeacock001) [inline HICPluginTests/Integration/ReflectionToDatabaseTester.cs:61]: There's a large space between length and 10, is that meant to be like that?

Comment (JFriel) [reply HICPluginTests/Integration/ReflectionToDatabaseTester.cs:61]: Looks like the way a double space in the exception message from a /loong/ time ago. Good catch!

**Review by jas88 — APPROVED** [review]

**Pull request merged**